### PR TITLE
make rubygems unit specs unit like

### PR DESCRIPTION
**Pull request by jaym**
These tests used to take 390ish seconds to run in appveyor

### CODE REVIEW

**Comment by ksubrama** [comment]
Not :construction: any more.  /cc @chef/client-windows

**Comment by ksubrama** [comment]
With this refactor, I have killed all the sequential state stored in instance variables like @gem_dep, @provider and @new_resource.  These are now changed to be very "rspec"-y, allowing each test to override values such as gem_name and bindir without having to re-initialize the provider over and over in each test.

The core speedup is seen in the `before` block of the "gem install" tests (line 527).  It stubs out "candidate_version_from_remote" and returns a preset version, allowing this test to bypass costly network operations.  The target version is set to 9000 just to further emphasize that the stub is effective - a real call out to rubygems would fail for such a version.

File: spec/unit/provider/package/rubygems_spec.rb [hunk 1]
```diff
@@ -327,357 +327,457 @@ def gemspec(name, version)
...
     end
 
     describe "when installing a gem" do
+      let(:target_version) { "9000.0.2" }
+      let(:current_version) { nil }
+      let(:candidate_version) { "9000.0.2" }
+      let(:current_resource) do
+        current_resource = Chef::Resource::GemPackage.new(gem_name)
+        current_resource.version(current_version)
+        #current_resource.gem_binary(gem_binary) if gem_binary
```
Comment (jaym) [inline spec/unit/provider/package/rubygems_spec.rb]: commented out code

**Comment by mwrock** [comment]
:+1: I like it. I also like to use rspec's `subject` for the actual "subject under test" like `provider` here but not a big deal. This is a nice improvement.

**Comment by jaym** [comment]
I don't have timings for how long this test now takes in appveyor, but it wasn't in the top 10, so it is significantly faster

:+1:

**Pull request merged**